token/driver/wallet.go

```diff
@@ -139,31 +139,19 @@ type WalletService interface {
 
 	// OwnerWallet returns an instance of the OwnerWallet interface bound to the passed id.
 	// The id can be: the wallet identifier or a unique id of a view identity belonging to the wallet.
-	OwnerWallet(id string) (OwnerWallet, error)
-
-	// OwnerWalletByIdentity returns the OwnerWallet the passed identity belongs to.
-	OwnerWalletByIdentity(identity view.Identity) (OwnerWallet, error)
+	OwnerWallet(id any) (OwnerWallet, error)
 
 	// IssuerWallet returns an instance of the IssuerWallet interface bound to the passed id.
 	// The id can be: the wallet identifier or a unique id of a view identity belonging to the wallet.
-	IssuerWallet(id string) (IssuerWallet, error)
-
-	// IssuerWalletByIdentity returns an instance of the IssuerWallet interface that contains the passed identity.
-	IssuerWalletByIdentity(identity view.Identity) (IssuerWallet, error)
-
-	// AuditorWalletByIdentity returns an instance of the AuditorWallet interface that contains the passed identity.
-	AuditorWalletByIdentity(identity view.Identity) (AuditorWallet, error)
+	IssuerWallet(id any) (IssuerWallet, error)
 
 	// AuditorWallet returns an instance of the AuditorWallet interface bound to the passed id.
 	// The id can be: the wallet identifier or a unique id of a view identity belonging to the wallet.
-	AuditorWallet(id string) (AuditorWallet, error)
+	AuditorWallet(id any) (AuditorWallet, error)
```

Comment:
Maybe it would be better to create another type instead of any, even if it is just an alias. So that we know what kind of `any` we expect to find.

Reply:
OK I can't find any way we can further restrict it or have some common functionality, but we could do the following:
* Try to replace also other points in code where this ID is passed (currently as `any` or `interface{}`, e.g. in `WalletRegistry.Lookup`, `Role.MapToID`)
* Write a comment above this type that the ID can be either `string` or `view.Identity`

Reply:
no, because it is the driver that must decide.